packages/strapi-plugin-i18n/services/locales.js

```diff
@@ -1,14 +1,51 @@
 'use strict';
 
-const find = (...args) => strapi.query('locale', 'i18n').find(...args);
+const { setIsDefault } = require('../utils');
 
-const findById = id => strapi.query('locale', 'i18n').findOne({ id });
+const getStore = () =>
+  strapi.store({
+    environment: '',
+    type: 'plugin',
+    name: 'i18n',
+  });
 
-const findByCode = code => strapi.query('locale', 'i18n').findOne({ code });
+const find = async (...args) => {
+  const locales = await strapi.query('locale', 'i18n').find(...args);
 
-const create = locale => strapi.query('locale', 'i18n').create(locale);
+  return setIsDefault(locales);
+};
+
+const findById = async id => {
+  const locale = await strapi.query('locale', 'i18n').findOne({ id });
+
+  return setIsDefault(locale);
+};
+
+const findByCode = async code => {
+  const locale = await strapi.query('locale', 'i18n').findOne({ code });
+
+  return setIsDefault(locale);
+};
 
-const update = (params, updates) => strapi.query('locale', 'i18n').update(params, updates);
+const create = async (locale, { isDefault }) => {
+  const createdLocale = await strapi.query('locale', 'i18n').create(locale);
+
+  if (isDefault) {
```

Comment:
Can you split the logic so you call a `setDefaultLocale` method from the controller instead so the locales servic stays focused on locales only.

Reply:
last change for that comment and we should be good to go